plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/NameEmbeddings.kt

```diff
@@ -117,8 +128,20 @@ fun FirFunctionSymbol<*>.embedTypeSignature(): String = valueParameterSymbols.jo
 fun FirValueParameterSymbol.embedName(): LocalName = LocalName(name, 0)
 
 fun FirPropertyAccessorSymbol.embedName(): MangledName {
-    val className = propertySymbol.callableId.classId!!.embedName()
+
     val propertyName = propertySymbol.callableId.callableName
+
+    // Extension properties do not use classId for the receiver type.
+    if (propertySymbol.callableId.classId == null) {
+        val className = propertySymbol.resolvedReceiverTypeRef!!.type.classId!!.embedName()
```

Comment:
This logic looks very ad-hoc.  Could it be split up a bit more so that we don't have to dig quite so deep into the internals of the symbol to find the right class?

Reply:
Since extension getters/setters are compiled down into static methods where the first parameter is the receiver type, I removed the reference to the class names. But, I kept the package name. See the latest commit for the [changes](https://github.com/jesyspa/kotlin/pull/109/commits/b80a50d1f593ab689c2499281052e302b1c1b03f).